File: backend/src/main/java/com/quisde/entity/Question.java
package com.quisde.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.List;
import java.util.ArrayList;

@Entity
@Table(name = "questions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "quiz_id", nullable = false)
    @JsonIgnore
    private Quiz quiz;

    @Column(nullable = false, length = 1024)
    private String text;

    @Column(nullable = false)
    private String type; // "multiple_choice", "true_false", "fill_blank"

    @Column(nullable = false)
    private String difficulty; // "easy", "medium", "hard"

    @Column(length = 512)
    private String explanation;

    @OneToMany(mappedBy = "question", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Answer> answers = new ArrayList<>();

    @Column(nullable = false)
    private boolean active = true;
} 

File: frontend/src/types/question.ts
export interface Question {
  id: number;
  text: string;
  type: string; // "multiple_choice", "true_false", "fill_blank"
  difficulty: string; // "easy", "medium", "hard"
  explanation?: string;
  answers: Answer[];
}

export interface Answer {
  id: number;
  text: string;
  isCorrect: boolean;
  explanation?: string;
  orderIndex: number;
}
